modular_skyrat/modules/microfusion/code/microfusion_gun_attachments.dm

```diff
@@ -0,0 +1,392 @@
+	name = "phase emitter heatsink"
+	desc = "Greatly increases the phase emitter cooling rate."
+	icon_state = "attachment_heatsink"
+	attachment_overlay_icon_state = "attachment_heatsink"
+	slot = GUN_SLOT_UNDERBARREL
+	/// Coolant bonus
+	var/cooling_rate_increase = 50
+
+/obj/item/microfusion_gun_attachment/heatsink/run_attachment(obj/item/gun/microfusion/microfusion_gun)
+	. = ..()
+	microfusion_gun.heat_dissipation_bonus += cooling_rate_increase
+
+/obj/item/microfusion_gun_attachment/heatsink/remove_attachment(obj/item/gun/microfusion/microfusion_gun)
+	. = ..()
+	microfusion_gun.heat_dissipation_bonus -= cooling_rate_increase
+
+/*
+UNDERCHARGER ATTACHMENT
+
+Massively decreases the output beam of the phase emitter.
+Converts shots to STAMNINA damage.
+*/
+/obj/item/microfusion_gun_attachment/undercharger
+	name = "phase emitter undercharger"
+	desc = "Inverts the output beam of the phase emitter."
+	icon_state = "attachment_undercharger"
+	attachment_overlay_icon_state = "attachment_undercharger"
+	slot = GUN_SLOT_UNDERBARREL
+	var/toggle = FALSE
+	var/cooling_rate_increase = 10
+
+/obj/item/microfusion_gun_attachment/undercharger/get_modify_data()
+	return list(list("title" = "Turn [toggle ? "OFF" : "ON"]", "icon" = "power-off", "color" = "[toggle ? "red" : "green"]", "reference" = "toggle_on_off"))
+
+/obj/item/microfusion_gun_attachment/undercharger/run_modify_data(params, mob/living/user, obj/item/gun/microfusion/microfusion_gun)
+	if(params == "toggle_on_off")
+		toggle(microfusion_gun, user)
+
+/obj/item/microfusion_gun_attachment/undercharger/proc/toggle(obj/item/gun/microfusion/microfusion_gun, mob/user)
+	if(toggle)
+		toggle = FALSE
+		microfusion_gun.heat_dissipation_bonus -= cooling_rate_increase
+	else
+		toggle = TRUE
+		microfusion_gun.heat_dissipation_bonus += cooling_rate_increase
+
+	if(user)
+		to_chat(user, span_notice("You toggle [src] [toggle ? "ON" : "OFF"]."))
+
+/obj/item/microfusion_gun_attachment/undercharger/run_attachment(obj/item/gun/microfusion/microfusion_gun)
+	. = ..()
+	microfusion_gun.fire_sound = 'modular_skyrat/modules/microfusion/sound/burn.ogg'
+
+/obj/item/microfusion_gun_attachment/undercharger/process_fire(obj/item/gun/microfusion/microfusion_gun, obj/item/ammo_casing/chambered)
+	. = ..()
+	if(toggle)
+		chambered.loaded_projectile?.damage_type = STAMINA
+		chambered.loaded_projectile?.icon_state = "disabler"
+		chambered.loaded_projectile?.light_color = COLOR_DARK_CYAN
+
+/obj/item/microfusion_gun_attachment/undercharger/remove_attachment(obj/item/gun/microfusion/microfusion_gun)
+	. = ..()
+	if(toggle)
+		microfusion_gun.heat_dissipation_bonus -= cooling_rate_increase
+	microfusion_gun.fire_sound = microfusion_gun.chambered?.fire_sound
+
+/*
+RGB ATTACHMENT
+
+Enables you to change the light color of the laser.
+*/
+/obj/item/microfusion_gun_attachment/rgb
+	name = "phase emitter spectrograph"
+	desc = "Enables the phase emitter to change it's output color."
```

Comment:
desc = "An attachment hooked up to the phase emitter, allowing the user to adjust the color of the beam outputted. This has seen widespread use by various factions capable of getting their hands on microfusion weapons, whether as a calling card or simply for entertainment."